CMakeLists.txt

```diff
@@ -63,293 +63,12 @@ CMAKE_DEPENDENT_OPTION(BUILD_RINGMESH_VIEW "Build ringmesh-view" ON
 
 set(CMAKE_INSTALL_MESSAGE LAZY)
 
-set(RINGMESH_EXTRA_ARGS
-    -DPROJECT_BINARY_DIR:STRING=${project_binary_dir_config}
-    -DMG_TETRA:STRING=${MG_TETRA}
-    -DRINGMESH_WITH_TETGEN:BOOL=${RINGMESH_WITH_TETGEN}
-    -DRINGMESH_WITH_GRAPHICS:BOOL=${RINGMESH_WITH_GRAPHICS}
-    -DBUILD_RINGMESH_VIEW:BOOL=${BUILD_RINGMESH_VIEW}
-    -DRINGMESH_WITH_UTILITIES:BOOL=${RINGMESH_WITH_UTILITIES}
-    -DRINGMESH_WITH_TESTS:BOOL=${RINGMESH_WITH_TESTS}
-    -DRINGMESH_WITH_TUTORIALS:BOOL=${RINGMESH_WITH_TUTORIALS}
-    -DBUILD_DOCUMENTATION:BOOL=${BUILD_DOCUMENTATION}
-    -DBUILD_GEOGRAM_WITHOUT_EXE:BOOL=${BUILD_GEOGRAM_WITHOUT_EXE})
-
-#------------------------------------------------------------------------------------------------
-# Generate configuration directories for single-configuration generators (Make)
-# and run cmake configuration command in each one of them
-if(CMAKE_GENERATOR STREQUAL "Unix Makefiles")
-    # If the CMAKE_BUILD_TYPE is no defined
-    # i.e. at first run of that file
-    if(NOT CMAKE_BUILD_TYPE)
-      # For all configuration types
-      foreach(config IN LISTS CMAKE_CONFIGURATION_TYPES)
-          project(MULTI_CONFIG NONE)
-          # The binary directory for this configuration
-          set(project_binary_dir_config  ${PROJECT_BINARY_DIR}/${config})
-
-          # Create the directory
-          file(MAKE_DIRECTORY ${project_binary_dir_config})
-
-          # Launch cmake for this configuration
-          # by specifying the CMAKE_BUILD_TYPE and the PROJECT _BINARY_DIR,
-          # and pass on all the options
-          execute_process(
-             COMMAND ${CMAKE_COMMAND} ${CMAKE_SOURCE_DIR}
-                -G ${CMAKE_GENERATOR}
-                ${RINGMESH_EXTRA_ARGS} 
-                -DCMAKE_BUILD_TYPE=${config}
-                -DGLOBAL_BINARY_DIR=${PROJECT_BINARY_DIR}
-                WORKING_DIRECTORY ${project_binary_dir_config})
-      endforeach()
-
-      # Get out when all configurations have been run
-      # We do not want to execute the rest of the file in that case
-      return()
-    endif()
-endif(CMAKE_GENERATOR STREQUAL "Unix Makefiles")
-
+# CMake is called, a main project (super build project) is generated first.
+# This main project is responsible to create third party cmake projects including
+# the one of RINGMesh. This is done during compilation, i.e., RINGMesh.cmake is
+# called (USE_SUPERBUILD = OFF) when the super build project is compiled.
 if(USE_SUPERBUILD)
-    # execute the superbuild
-    project(SUPERBUILD)
-    if(NOT GLOBAL_BINARY_DIR)
-        set(GLOBAL_BINARY_DIR ${PROJECT_BINARY_DIR})
-    endif()
     include("${CMAKE_SOURCE_DIR}/cmake/SuperBuild.cmake")
```

Comment:
You could simplify the path:  include("${CMAKE_SOURCE_DIR}/cmake/SuperBuild.cmake") = >  include(cmake/SuperBuild.cmake)